huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go

```diff
@@ -0,0 +1,574 @@
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"public_ip": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Optional: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"eip_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+							Optional: true,
+							ForceNew: true,
+						},
+						"public_bind_type": {
+							Type:     schema.TypeString,
+							Computed: true,
+							Optional: true,
+							ForceNew: true,
+						},
+					},
+				},
+			},
+
+			"number_of_cn": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"enterprise_project_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+
+			"tags": common.TagsForceNewSchema(),
+
+			"created": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"endpoints": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"connect_info": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"jdbc_url": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"public_endpoints": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"jdbc_url": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"public_connect_info": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"recent_event": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"sub_status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"task_status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"updated": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"private_ip": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func resourceDwsClusterCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+	client, err := config.DwsV1Client(region)
+	if err != nil {
+		return fmtp.DiagErrorf("error creating DWS v1 client, err=%s", err)
+	}
+
+	opts := &cluster.CreateOpts{
+		Name:                d.Get("name").(string),
+		AvailabilityZone:    d.Get("availability_zone").(string),
+		SubnetID:            d.Get("network_id").(string),
+		NodeType:            d.Get("node_type").(string),
+		UserName:            d.Get("user_name").(string),
+		UserPwd:             d.Get("user_pwd").(string),
```

Comment:
pls do not set the password field before printing opts to the log file

Reply:
Ok, remove the log.printf